src/nvim/os/env.c

```diff
@@ -243,19 +243,24 @@ void expand_env_esc(char_u *srcp, char_u *dst, int dstlen, bool esc, bool one,
         // Verify that we have found the end of a UNIX ${VAR} style variable
         if (src[1] == '{' && *tail != '}') {
           var = NULL;
-        } else if (src[1] == '{') {
-          ++tail;
-        }
+        } else {
+          if (src[1] == '{') {
+            ++tail;
+          }
 #elif defined(MSWIN)
```

Comment:
This `#elif ... #endif` could be removed completely, because `src[0]` is never '%'. In vim this code was useful, because of this: https://github.com/vim/vim/blob/master/src/misc1.c#L3974-L3976